core/src/test/java/io/confluent/rest/ApplicationGroupTest.java

```diff
@@ -116,6 +116,20 @@ protected ResourceCollection getStaticResources() {
     assertThat(makeGetRequest("/app2/index.html"), is(Code.OK));
   }
 
+  /* Test Dynamic App Loading */
+  @Test
+  public void testDynamicAppLoading() throws Exception {
+    TestApp app1 = new TestApp("/app1");
+    TestApp app2 = new TestApp("/app2");
+
+    server.registerApplication(app1);
+    server.start();
+    server.registerApplication(app2);
+
+    assertThat(makeGetRequest("/app1/resource"), is(Code.OK));
+    assertThat(makeGetRequest("/app2/resource"), is(Code.OK));
```

Comment:
just to be really pedantic I would also want to verify that things aren't started early.

```
server.registerApplication(app1);
// call and get no response from app1

 server.start();
    assertThat(makeGetRequest("/app1/resource"), is(Code.OK));
// verify app2 does not start

    server.registerApplication(app2);

    // app2 is not good
    assertThat(makeGetRequest("/app2/resource"), is(Code.OK));
    // app 2 still works
    assertThat(makeGetRequest("/app1/resource"), is(Code.OK));
```